esmvaltool/diag_scripts/climate_patterns/climate_patterns.py

```diff
@@ -0,0 +1,658 @@
+    )
+
+    return clim_list_final, ts_list_final
+
+
+def append_diurnal_range(derived_diurnal_clim,
+                         derived_diurnal_ts,
+                         clim_list,
+                         ts_list):
+    """Append diurnal range to cubelists.
+
+    Parameters
+    ----------
+    derived_diurnal_clim : cube
+        derived diurnal climatology cube
+    derived_diurnal_ts : cube
+        derived diurnal timeseries cube
+    clim_list : cubelist
+        existing climatology cubelist, no range
+    ts_list : cubelist
+        existing timeseries cubelist, no range
+
+    Returns
+    -------
+    clim_list_final : cubelist
+        cubelist of climatology cubes including diurnal range
+    ts_list_final : cubelist
+        cubelist of standard timeseries cubes including diurnal range
+    """
+    # creating cube list without tasmax or tasmin
+    # (since we just wanted the diurnal range)
+    clim_list_final = iris.cube.CubeList([])
+    ts_list_final = iris.cube.CubeList([])
+
+    for cube in clim_list:
+        if cube.var_name not in ("tasmax", "tasmin"):
+            clim_list_final.append(cube)
+
+    for cube in ts_list:
+        if cube.var_name not in ("tasmax", "tasmin"):
+            ts_list_final.append(cube)
+
+    clim_list_final.append(derived_diurnal_clim)
+    ts_list_final.append(derived_diurnal_ts)
+
+    return clim_list_final, ts_list_final
+
+
+def calculate_anomaly(clim_list, ts_list):
+    """Calculate variables as anomalies, and adds diurnal range as variable.
+
+    Parameters
+    ----------
+    clim_list : cubelist
+        cubelist of climatology variables
+    ts_list : cubelist
+        cubelist of standard variable timeseries
+
+    Returns
+    -------
+    clim_list_final : cubelist
+        cubelist of clim. vars, inc. diurnal range
+    anom_list_final : cubelist
+        cubelist of anomaly vars, inc. diurnal range
+    """
+    # calculate diurnal temperature range cube
+    clim_list_final, ts_list_final = calculate_diurnal_range(
+        clim_list,
+        ts_list
+    )
+
+    anom_list_final = ts_list_final.copy()
+
+    # calc the anom by subtracting the monthly climatology from
+    # the time series
+    for i, _ in enumerate(ts_list_final):
+        i_months = (
+            anom_list_final[i].coord("month_number").points - 1
+        )  # -1 because months are numbered 1..12
+        anom_list_final[i].data -= clim_list_final[i][i_months].data
+
+    return clim_list_final, anom_list_final
+
+
+def regression(tas, cube_data, area, ocean_frac=None, land_frac=None):
+    """Calculate coeffs of regression between global surf temp and variable.
+
+    Parameters
+    ----------
+    tas : cube
+        near-surface air temperature
+    cube_data : arr
+        cube.data array of a variable
+    ocean_frac: cube
+        gridded ocean fraction
+    land_frac: cube
+        gridded land fraction
+    area: str
+        area over which to calculate patterns
```

Comment:
Would it be possible to ensure the order of these parameters matches with the order they are defined in the function, please? 😊

Reply:
Resolved in the latest commit :)